chains/ethereum/itx_client.go

```diff
@@ -0,0 +1,148 @@
+package ethereum
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"strings"
+
+	"github.com/ChainSafe/ChainBridge/bindings/Bridge"
+	"github.com/ChainSafe/chainbridge-utils/crypto/secp256k1"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/rpc"
+)
+
+type RelayTx struct {
+	to       common.Address
+	data     []byte
+	gasLimit uint
+	schedule string
+}
+
+// ABI encode the vote proposal function args
+func packVoteProposalData(chainID uint8, depositNonce uint64, resourceID [32]byte, dataHash [32]byte) ([]byte, error) {
+	parsed, err := abi.JSON(strings.NewReader(Bridge.BridgeABI))
+	if err != nil {
+		return nil, err
+	}
+	packed, err := parsed.Pack("voteProposal", chainID, depositNonce, resourceID, dataHash)
+	if err != nil {
+		return nil, err
+	}
+	return packed, nil
+}
+
+// ABI encode the execute proposal function args
+func packExecuteProposalData(chainID uint8, depositNonce uint64, data []byte, resourceID [32]byte) ([]byte, error) {
+	parsed, err := abi.JSON(strings.NewReader(Bridge.BridgeABI))
+	if err != nil {
+		return nil, err
+	}
+	packed, err := parsed.Pack("executeProposal", chainID, depositNonce, data, resourceID)
+	if err != nil {
+		return nil, err
+	}
+	return packed, nil
+}
+
+// Form and sign a relay transaction
+func toSignedRelayTx(to string, data []byte, gasLimit uint, chainId uint, keyPair *secp256k1.Keypair) (*struct {
+	tx   RelayTx
+	sig  []byte
+	txId string
+}, error) {
+	// see https://infura.io/docs/transactions#section/Developing-with-ITX/Signing-a-relay-request
+	toAddress := common.HexToAddress(to)
+	gasInt := big.NewInt(int64(gasLimit))
+	chainIdBigInt := big.NewInt(int64(chainId))
+	schedule := "fast"
+
+	uint256Ty, err := abi.NewType("uint256", "uint256", nil)
+	if err != nil {
+		return nil, err
+	}
+	addressTy, err := abi.NewType("address", "address", nil)
+	if err != nil {
+		return nil, err
+	}
+	bytesTy, err := abi.NewType("bytes", "bytes", nil)
+	if err != nil {
+		return nil, err
+	}
+	stringTy, err := abi.NewType("string", "string", nil)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Some linters would tell you to give these `if err != nil {...}` blocks room to breath: https://github.com/bombsimon/wsl/blob/master/doc/rules.md#assignments-should-only-be-cuddled-with-other-assignments

Reply:
Split up the cuddlers :hugs: